### ISSUE TITLE
Completion does not seem to work

### ISSUE THREAD

**Issue by brucestephens** [Andersbakken/rtags#127]
With the following code (with the cursor at the end of the last `foo.` line), I was hoping `rtags-code-complete-at` would suggest the two field names.  Or at least do something, but it seems not to do anything.

`rc --code-completion-enabled` reports 1, and the project is loaded and other things work (like `rtags-find-symbol-at-point`).

```
struct Foo {
  int field_1;
  int another_field_2;
};

void one()
{
  Foo foo;
  foo.field_1 = 2;
  foo.
}
```

(Calling it by hand also seems not to do what I expect, but by the looks of it it's waiting for input from `stdin`, so likely that's expected.)

**Comment by stephenry** [comment]
I too am having some considerable difficulty in getting completion to work using rtags. I have the following code in my init.el. I'm not in a place to debug because after "." or "->" nothing happens. Can any one provide any suggestions?

(require 'auto-complete)
(require 'auto-complete-config)
(ac-config-default)

(require 'rtags)
(require 'popup)
(require 'rtags-ac)
(rtags-enable-standard-keybindings c-mode-base-map)
(setq rtags-completions-enabled t)

(add-hook 'c-mode-common-hook
	  (lambda ()
	    (setq ac-sources '(ac-sources-rtags))))

**Comment by Mafeuser** [comment]
Hi Anders.
I'm running rtags since 2 weeks. This is real charm!
I'm also trying to run completion. No success here unfortunatelly.
Today I had session with my debugger and have some conclusions.

My setup is:
lubuntu.14.04, gcc.4.8.2/clang.3.5/rtags:d1a4728

trying to complete very simple C++ program. Jumping between definition/reference works correct.
But when trying to complete on [POINT] I get no result at all

#include <string>
class CLS
{
  public:
    CLS();
    std::string name() { return "CLS"; }
    void foo() { this->nam[POINT]
    virtual ~CLS();
};

int main(int argc, char *argv[]) {  return 0; }

emacs for sure calls rdm through rc. My *RTags Log* tells this:
/usr/local/bin/rc --current-file=/home/user/cxx/p0.cpp --unsaved-file=/home/user/cxx/p0.cpp:193 -Y -l /home/user/cxx/p0.cpp:7:24:

but rdm backend is silent about the line above. Looks like it never receives the request.

debugger on rc binary leads me here:
Breakpoint 1, RClient::parse (this=this@entry=0xbfa4d4ec, 
    argc=@0xbfa4d590: 6, argv=argv@entry=0xbfa4d624)
    at /home/user/rtags/src/RClient.cpp:658
658	            std::regex rx("^(.*):([0-9]+):([0-9]+):?");
657	            std::cmatch caps;
658	            std::regex rx("^(.*):([0-9]+):([0-9]+):?");
657	            std::cmatch caps;
658	            std::regex rx("^(.*):([0-9]+):([0-9]+):?");

Program received signal SIGABRT, Aborted.
0xb76fc424 in ?? ()
 
Is it the problem with std::regex implementation present in gcc.4.8.2?
what gcc version do YOu use?

http://stackoverflow.com/questions/8060025/is-this-c11-regex-error-me-or-the-compiler

best regards,
Paul

**Comment by pradhanpk** [comment]
@stephenry I switched to `company-mode` after seeing Sohail's screenshot above and remember very little about `auto-complete` now. However, if you are fine with using `company-mode`, I would be happy to help you out.

**Comment by Mafeuser** [comment]
If std::regex is the case, both the completion methods are affected as far as gcc lower than 4.9.1 is used.
the cure would probably be to upgrade to gcc 4.9.1 (not alwas possible) or to use boost::regex instead.

**Comment by Mafeuser** [comment]
I just upgraded gcc from version 4.8.2 to 4.9.2 and completion started working.

**Comment by mawww** [comment]
gcc 4.8 regex implementation is stubbed, it does not do anything, complete std::regex implementation is only available since gcc 4.9.
